src/env_node.ml

```diff
@@ -109,3 +113,53 @@ let rec ocaml_flags t ~profile ~expander =
     in
     t.ocaml_flags <- Some flags;
     flags
+
+
+
+let rec c_flags t ~profile ~expander =
+  match t.c_flags with
+  | Some x -> x
+  | None ->
+    let default =
+      match t.inherit_from with
+      | None -> Build.return ([])
+      | Some (lazy t) -> c_flags t ~profile ~expander
+    in
+    let flags =
+      match find_config t ~profile with
+      | None -> default
+      | Some cfg ->
+        let expander = Expander.set_dir expander ~dir:t.dir in
+        let eval = Expander.expand_and_eval_set expander in
+        let f = cfg.c_flags in
+        if  Ordered_set_lang.Unexpanded.has_special_forms f then
+          eval f ~standard:default
+        else
+          eval f ~standard:(Build.return [])
+    in
+    t.c_flags <- Some flags;
+    flags
+
+let rec cxx_flags t ~profile ~expander =
```

Comment:
This function looks awfully similar to `c_flags`. Would you mind trying to generalize them to with some sort of `make_flags` function? It would probably require passing some functions as parameters but it's worth it in this case I think.

Reply:
This kind of recursive search through directories is a recurrent pattern in env_node and it is used by other existing functions (ocaml_flags, local_binaries, artifcats).  I am not sure how to best fix it. Wouldn't it be better to fix this in a separate PR?